src/Symfony/Component/HttpKernel/DataCollector/LoggerDataCollector.php

```diff
@@ -181,4 +211,33 @@ private function computeErrorsCount()
 
         return $count;
     }
+
+    public function persistContainerBuildLogs()
+    {
+        $datas = [
+            'container_logs' => count($this->containerBuildLogs),
+            'logs' => $this->containerBuildLogs
+        ];
+
+        $this->data['container_build_logs']  = $this->sanitizeLogs($this->containerBuildLogs);
+
+        $this->filesystem->dumpFile($this->getContainerBuildLogsFilename(), serialize($datas));
+    }
+
+    /**
+     * @return string
+     */
+    public function getContainerBuildLogsFilename()
+    {
+        return $this->cacheDir.'/containerBuildLogs';
+    }
+
+    public function loadContainerBuildLogs()
+    {
+        $rawContent = file_get_contents($this->getContainerBuildLogsFilename());
+        $datas = unserialize($rawContent);
+
+        $this->data['container_build_count'] = $datas['container_logs'];
+        $this->data['container_build_logs']  = $this->sanitizeLogs($datas['logs']);
```

Comment:
sanitizing must be done before dumping. Otherwise, you have no guarantee that the logs will be serializable (the context can contain objects)